lib/run-compilation.js

```diff
@@ -46,7 +46,12 @@ module.exports = function runCompilation (options, compiler, done) {
     }
 
     if (errors.length) {
-      compilation.errors.push(chalk.red(options.formatter(errors)));
+      if (options.emitErrors === false) {
+        var suppressionWarning = 'Stylelint error, reported as warning because emitErrors is set to false';
+        compilation.warnings.push(chalk.yellow(options.formatter(errors + ' - ' + suppressionWarning)));
```

Comment:
looks like `errors` (the thing passed to style lint's default formatter) is meant to be an array, so you should probably use `errors.concat(suppressionWarning)` or something to that effect.